src/chars.rs

```diff
@@ -128,15 +128,29 @@ pub(crate) fn count_impl<T: ByteChunk>(text: &[u8]) -> usize {
             inv_count += ((byte & 0xC0) == 0x80) as usize;
         }
 
-        // Take care of the middle bytes in big chunks.
-        for chunks in middle.chunks(T::MAX_ACC) {
-            let mut acc = T::zero();
-            for chunk in chunks.iter() {
-                acc = acc.add(chunk.bitand(T::splat(0xc0)).cmp_eq_byte(0x80));
+        let char_boundaries = |val: &T| val.bitand(T::splat(0xc0)).cmp_eq_byte(0x80);
```

Comment:
This might be better as an inner function rather than a closure.  Stylistically, I don't mind either way.  But changing it to an inner function means we can put `#[inline(always)]` on it to ensure that it always gets inlined.

(It almost certainly doesn't actually make a difference, but just to not be at the mercy of the compiler.  Particularly since we're already jumping through the hoop of explicit loop unrolling for similar reasons.)